WalletWasabi/Wallets/Wallet.cs

```diff
@@ -231,6 +231,11 @@ public HdPubKey GetNextReceiveAddress(IEnumerable<string> destinationLabels)
 
 	public int GetPrivacyPercentage()
 	{
+		if (!Coins.Any())
+		{
+			return 0;
+		}
+
```

Comment:
@ichthus1604 This is ok, but a better solution would be to remove your changes and modify this condition in L241:

```
maxPrivacyScore == 0M ? 100
```
To this:
```
maxPrivacyScore == 0M ? 0
```

##

Also this percentage calculation is used by Coinjoin so I am not sure if the change is safe. @molnard What do you think?

Reply:
@soosr I don't think this is entirely correct.

Let me explain my reasoning:

this line:
```
var maxPrivacyScore = Coins.TotalAmount().Satoshi * (AnonScoreTarget - 1);
```

will return 0 if your `AnonScoreTarget = 1`, regardless of the number of coins in the wallet.

It makes sense that if your `AnonScoreTarget = 1` for whatever reason, then you have a privacy score of 100%, unless of course you have no coins, in which case my code kicks in and you have 0% (or really N/A, but for some reason the trigger is fired multiple times and the UI receives 0% as a valid value from the business code)

With this, I believe it is correct that we show 100% for a wallet with > 0 coins and AnonScoreTarget = 1, and 0% for a wallet with no coins at all.

What do you think?

Reply:
`AnonScoreTarget` can never be 1. The minimum is 2. So you don't have to deal with it.

Reply:
@soosr Understood. Applied your suggestion.